### ISSUE TITLE
Pod running on windows node with containerd runtime stuck at PodInitializing

### ISSUE THREAD

**Issue by robinshine** [kubernetes/kubernetes#104702]
#### What happened:

I have a pod containing an init container and main container running on Windows node with containerd runtime. The init container completes successfully, while the main container stuck at PodInitializing. 

#### What you expected to happen:

The main container should start to run when init container completes successfully, just like what happens when using the docker runtime. 

#### How to reproduce it (as minimally and precisely as possible):

1. Create a GKE standard cluster from rapid release channel, using version 1.21.3-gke.2000
2. Add a new pool with image set to "Windows Long Term Servicing Channel with Containerd (windows_ltsc_containerd)
3. Create the cluster and configure kubectl to connect to the cluster
4. Create a test namespace by running `kubectl create namespace test`
5. Create a file `test.yaml` with below content:
    ```
    apiVersion: v1
    kind: Pod
    metadata: {name: job, namespace: test}
    spec:
      containers:
      - args: [/c, echo, main]
        image: mcr.microsoft.com/windows/servercore:1809
        name: main
        command: [cmd]
      initContainers:
      - args: [/c, echo, init]
        image: mcr.microsoft.com/windows/servercore:1809
        name: init
        command: [cmd]
      restartPolicy: Never
      nodeSelector: {kubernetes.io/os: windows}
    ```
6. Run command `kubectl apply -f test.yml` to create the pod into test namespace
7. Run command `kubectl describe pod job -n test` to check pod status. The init container will complete successfully after a while, while main container stuck at waiting state with reason "PodInitializing"

#### Anything else we need to know?:

This issue does not exist if Windows image with docker runtime is used. Also works fine if run the pod on Linux node with containerd runtime.

#### Environment:
- Kubernetes version (use `kubectl version`): 1.21.3-gke.2000
- Cloud provider or hardware configuration: GKE standard
- OS (e.g: `cat /etc/os-release`): 
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

**Comment by vboulineau** [comment]
We've encountered the same issue while deploying on Windows+Containerd, I gathered debug `containerd` logs + some process events here: https://gist.github.com/vboulineau/74a84060ad59758f95ff40d3f13c721f

From that, I think maybe this is because the PodSandbox shim exits (`eb8c0244e080b1a25fba3e89949679cc104c0dbfa8ed92efb005e0f1eea7ebab`) after the init container finishes (`38802092e14f81ec0d372518bb27d4627482ee25e6b39aa117eca58e711d0220`).

I tried to check `containerd` and `runhcs` code but I'm not sure on which side the issue lies as the *rpc flows are not straightforward to follow.

**Comment by ibabou** [comment]
Thanks Robin/Vincent for the inputs! I was able to repro the same behavior on my side.

This seems related to an issue with containerd 1.5.2 bins (which is used on the cluster 1.21.3-gke.2000 mentioned above) - causing the shim to exit if no containers running, that have been addressed with our containerd 1.5.4 bits. The upcoming GKE 2021-R30 (v1.21.4-gke.2300) release already includes the containerd 1.5.4 bits as the runtime on Windows Containerd pools (WINDOWS_LTSC_CONTAINERD & WINDOWS_SAC_CONTAINERD).
